### PR TITLE
Feature/data explorer metadata

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR shows the metadata depending on the selected source:

- Fetch methodology metadata
- 'Raw Data - Methodology' menu with hashes
- Fetch metadata for the different filters depending on the section
- Create selectors and change URL when the selection changes
- Select metadata depending on the selected source

Note: Still no filtering to the data is applied

image

### CODE REVIEW

**Comment by simaob** [comment]
Not sure where the source content is coming from, but there should be a change, that maybe we won't do here only but needs to be a bit broader.

As per this thread: https://basecamp.com/1756858/projects/13795275/todos/348881975#comment_614177959, the way the sources should be organised is:

CAIT (AR2)
PIK - AR4
PIK - AR2
UNFCCC - Annex I (AR4)
UNFCCC - non Annex I (AR2)

But this I believe will require a reorganising of the backend data.

**Comment by simaob** [comment]
Other than that the metadata is showing =)

File: app/javascript/app/utils/utils.js [hunk 1]
```diff
@@ -137,6 +137,40 @@ export const wordWrap = (long_string, max_char) => {
   return split_out;
 };
 
+const unquote = value => {
+  if (value.charAt(0) === '"' && value.charAt(value.length - 1) === '"') { return value.substring(1, value.length - 1); }
+  return value;
+};
+
+export const parseLinkHeader = header => {
```
Comment (weberjavi) [inline app/javascript/app/utils/utils.js]: 😱

File: app/javascript/app/providers/data-explorer-provider/data-explorer-provider-actions.js [hunk 2]
```diff
@@ -40,9 +56,100 @@ export const fetchDataExplorer = createThunkAction(
...
+        }
+      })
+      .catch(error => {
+        console.warn(error);
+        dispatch(fetchSectionMetadataFail());
+      });
+  }
+);
+
+export const fetchMetadata = createThunkAction(
```
Comment (weberjavi) [inline app/javascript/app/providers/data-explorer-provider/data-explorer-provider-actions.js]: Quite a complex fetching staff going on. Hope you didn't lose the nerve 😅

File: app/javascript/app/utils/utils.js (hunk 1 above)
Comment (weberjavi) [reply app/javascript/app/utils/utils.js]: good old Regex

**Comment by simaob** [comment]
I've created a different story on pivotal tracker to look at the sources more generally, so I think this can be merged now!

**Pull request merged**